mesh/mesh.go

```diff
@@ -518,12 +518,13 @@ func (msh *Mesh) applyState(ctx context.Context, logger log.Log, lid types.Layer
 	block := msh.getBlockToApply(valids)
 	if block != nil {
 		applied = block.ID()
-		err := msh.conState.ApplyLayer(ctx, block)
-		if err != nil {
-			logger.With().Error("failed to apply transactions", log.Err(err))
-			return fmt.Errorf("apply layer: %w", err)
-		}
 	}
+
+	if err := msh.conState.ApplyLayer(ctx, lid, block); err != nil {
+		logger.With().Error("failed to apply transactions", log.Err(err))
+		return fmt.Errorf("apply layer: %w", err)
```

Comment:
I know the code was just moved, but it's a great moment to improve :) 

IMO errors should be logged when they are handled because otherwise they might be logged multiple times (every time they are bubbled up). The attached context should contain all extra information required to debug.
```suggestion
		return fmt.Errorf("failed to apply transactions for layer %d: %w", lid.Value, err)
```

Reply:
> IMO errors should be logged when they are handled because otherwise they might be logged multiple times (every time they are bubbled up). The attached context should contain all extra information required to debug.

+1. they are logged multiple times and code looks messy when logs are written in this style

Reply:
done for mesh.go